cmd/otel-allocator/watcher/promOperator.go

```diff
@@ -37,13 +37,13 @@ import (
 	allocatorconfig "github.com/open-telemetry/opentelemetry-operator/cmd/otel-allocator/config"
 )
 
-func NewPrometheusCRWatcher(logger logr.Logger, cfg allocatorconfig.Config, cliConfig allocatorconfig.CLIConfig) (*PrometheusCRWatcher, error) {
-	mClient, err := monitoringclient.NewForConfig(cliConfig.ClusterConfig)
+func NewPrometheusCRWatcher(logger logr.Logger, cfg *allocatorconfig.Config) (*PrometheusCRWatcher, error) {
```

Comment:
Should we check for nil here?

Reply:
I think I'd prefer to just pass it by value. I don't think that's more expensive in a meaningful way, and this is clearly not supposed to be nil.